### PR TITLE
Issue #2464 Addon: Specify empty value for Var-Defaults

### PR DESCRIPTION
**Pull request by praveenkumar**
How to test
--------------
```
$ cat ~/.minishift/addons/test/test.addon
# Name: test
# Description: Test.
# Required-Vars: FOO
# Var-Defaults: FOO=null

echo Out of 'FOO'
echo #{FOO}

$ ./minishift addon apply test
-- Applying addon 'test':
Out of 'FOO'

```

### CODE REVIEW

File: pkg/minishift/addon/addon_meta.go [hunk 1]
```diff
@@ -95,6 +96,10 @@ func (meta *DefaultAddOnMeta) VarDefaults() ([]RequiredVar, error) {
 		defaults := make([]RequiredVar, 0, len(items))
 		for _, item := range items {
 			varDefault, _ := minishiftStrings.SplitAndTrim(item, "=")
+			// In case of value of varDefault is null/Null/NULL then value converted to empty string.
+			if strings.ToLower(varDefault[1]) == "null" {
+				varDefault[1] = ""
+			}
```
Comment (praveenkumar) [inline pkg/minishift/addon/addon_meta.go]: https://github.com/minishift/minishift/issues/2464#issuecomment-400266767

**Comment by praveenkumar** [comment]
retest this please.

File: pkg/minishift/addon/addon_meta.go (hunk 1 above)
Comment (gbraad) [reply pkg/minishift/addon/addon_meta.go]: not sure if I like to see `null`... or rather `nil`.
nil has a clearer meaning in language than `null`, as the last one means related to zero or invalid, while nil means non-existant.

does it actually NEED a value? why is `FOO=` not possible?

Comment (praveenkumar) [reply pkg/minishift/addon/addon_meta.go]: > nil has a clearer meaning in language than null, as the last one means related to zero or invalid, while nil means non-existant.

@gbraad sure that something we can modify

> does it actually NEED a value? why is FOO= not possible?

@gbraad https://github.com/minishift/minishift/issues/2464#issuecomment-400266767

Comment (LalatenduMohanty) [reply pkg/minishift/addon/addon_meta.go]: Both `nil` or `null` works for me.

**Comment by LalatenduMohanty** [comment]
ok to test

**Pull request merged**